Add item requests

87stels87 opened the pull request.

87stels87 committed:
14 спринт. попытка 1

87stels87 committed:
15 спринт. попытка 1

87stels87 committed:
15 спринт. попытка 2 (увеличил покрытие строк)

Sla-als commented on src/main/java/ru/practicum/shareit/booking/BookingController.java:
```diff
@@ -20,45 +21,49 @@ public class BookingController {
     @GetMapping("/{bookingId}")
-    public BookingOutputDto bookingOutputDto(@PathVariable Long bookingId,
-                                             @RequestHeader("X-Sharer-User-Id") Long userId) {
+    public ResponseEntity<BookingOutputDto> bookingOutputDto(@PathVariable Long bookingId,
```
следует избегать таких неясных названий для методов. Метод должен быть назван в соответствии с тем, что он делает, например, getBookingById.

Sla-als commented on src/main/java/ru/practicum/shareit/request/RequestMapper.java:
```diff
@@ -0,0 +1,37 @@
+        return itemRequest;
+    }
+
+//    public static List<ItemRequestDto> returnItemRequestDtoList(Iterable<ItemRequest> requests) {
```
Это нужно удалить

Sla-als commented on src/main/resources/application.properties:
```diff
@@ -2,12 +2,14 @@ spring.jpa.hibernate.ddl-auto=none
-
-logging.level.org.springframework.orm.jpa=INFO
-logging.level.org.springframework.transaction=INFO
+#logging.level.org.springframework.orm.jpa=INFO
```
Закомментированные строки нужно удалить

Sla-als commented on src/main/java/ru/practicum/shareit/request/ItemRequestServiceImpl.java:
```diff
@@ -0,0 +1,92 @@
+@Service
+@Transactional(readOnly = true)
+@RequiredArgsConstructor
+public class ItemRequestServiceImpl implements ItemRequestService {
```
В методах, где выполняются операции с репозиторием, лучше обрабатывать отсутствующие данные с использованием Optional вместо вызова .get(), чтобы избежать возможных NoSuchElementException

Sla-als commented on src/test/java/ru/practicum/shareit/ShareItTests.java:
```diff
@@ -6,8 +6,8 @@
-	@Test
-	void contextLoads() {
-	}
+    @Test
```
?

Sla-als commented on src/test/java/ru/practicum/shareit/helpers/HelpersServiceTest.java:
```diff
@@ -0,0 +1,66 @@
+    private ItemRequestRepo itemRequestRepo;
+
+    @Test
+    void testUser() {
```
Методы теста следует именовать более осмысленно. Например, вместо testUser лучше использовать shouldThrowExceptionWhenUserNotFound.

Sla-als commented on src/test/java/ru/practicum/shareit/item/ItemControllerTest.java:
```diff
@@ -0,0 +1,201 @@
+                .email("andrey@yandex.ru")
+                .build();
+
+        itemRequest = ItemRequest.builder()
```
В DTO itemDto1 и itemDto2 у вас одинаковый идентификатор (id = 1L). Это может вызвать путаницу в тестах и делает ваши данные менее реалистичными. Лучше изменить один из идентификаторов

Sla-als commented on src/test/java/ru/practicum/shareit/item/ItemControllerTest.java:
```diff
@@ -0,0 +1,201 @@
+
+        when(itemService.getItemsByUserId(anyLong(), anyInt(), anyInt())).thenReturn(List.of(itemDto1, itemDto2));
+
+        mvc.perform(get("/items")
```
В тестовых методах есть повторяющийся код, связанный с выполнением запросов через mvc.perform(). Рассмотрите возможность вынесения этого кода в отдельные вспомогательные методы, чтобы сделать тесты более читаемыми.

Sla-als commented on src/test/java/ru/practicum/shareit/item/ItemControllerTest.java:
```diff
@@ -0,0 +1,201 @@
+
+
+@WebMvcTest(controllers = ItemController.class)
+public class ItemControllerTest {
```
Вместо проверки каждого поля в JSON-ответе можно сериализовать весь объект DTO в строку JSON и сравнивать всю строку с ожидаемым результатом.

Sla-als reviewed (commented):
Здравствуйте! Ваш код в целом выглядит неплохо, но есть несколько замечаний и предложений по улучшению

87stels87 committed:
15 спринт. исправления по результатам ревью
- переименовал метод в BookingController.java на getBookingById

87stels87 committed:
15 спринт. исправления по результатам ревью
- удалил закомментированный участок кода в RequestMapper
- удалил закомментированный участок кода в application.properties

87stels87 committed:
15 спринт. исправления по результатам ревью
- в классе ItemControllerTest изменил айдишник у  itemDto2

87stels87 committed:
15 спринт. исправления по результатам ревью
- в классе HelpersServiceTest изменил наименования тестов используя паттерн should_when

Sla-als commented:
Вы отлично поработали на этом спринте!
